optimize-data-indexing/OptimizeDataIndexing/Program.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Search;
+using Microsoft.Azure.Search.Models;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Index = Microsoft.Azure.Search.Models.Index;
+
+namespace OptimizeDataIndexing
+{
+    class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            IConfigurationBuilder builder = new ConfigurationBuilder().AddJsonFile("appsettings.json");
+            IConfigurationRoot configuration = builder.Build();
+
+            SearchServiceClient serviceClient = CreateSearchServiceClient(configuration);
+
+            string indexName = configuration["SearchIndexName"];
+            ISearchIndexClient indexClient = serviceClient.Indexes.GetClient(indexName);
+
+            Console.WriteLine("{0}", "Deleting index...\n");
+            await DeleteIndexIfExists(indexName, serviceClient);
+
+            Console.WriteLine("{0}", "Creating index...\n");
+            await CreateIndex(indexName, serviceClient);
+
+            Console.WriteLine("{0}", "Finding optimal batch size...\n");
```

Comment:
So this code is going to suggest a batch size??  wow that is cool, and different from what I would have guessed.  Not being sure, I don't want to presume that in the readme edits, but if your code is suggesting a batch size, I would try to explain that a little bit.